src/vma/iomux/io_mux_call.cpp

```diff
@@ -245,37 +245,54 @@ void io_mux_call::check_offloaded_rsockets(uint64_t *p_poll_sn)
 	//return false;
 }
 
+bool io_mux_call::handle_os_countdown(int &poll_os_countdown)
+{
+	/*
+	 * Poll OS when count down reaches zero. This honors CQ-OS ratio.
+	 * This also handles the 0 ratio case - do not poll OS at all.
+	 */
+	if (poll_os_countdown-- == 0 && m_n_sysvar_select_poll_os_ratio > 0) {
+		bool cq_ready = wait_os(true);
+		if (cq_ready) {
```

Comment:
Please consider to move wait_os() inside if statement.

Reply:
:+1: